hash.c

```diff
@@ -6736,9 +6906,25 @@ env_update(VALUE env, VALUE hash)
  *
  *  You can set the default proc to +nil+, which restores control to the default value:
  *
+ *    h.delete(:nosuch)
  *    h.default_proc = nil
  *    h.default = false
  *    h[:nosuch] # => false
+ *
+ *  === Chaining \Method Calls
+ *
+ *  Some \Hash instance methods return <tt>self</tt>.
+ *  For those methods, you can "chain" method calls
+ *  by following one method call with another.
+ *
+ *  This example chains methods #merge! and #compact!:
+ *
+ *  - <tt>merge</tt> merges another \Hash into a copy of <tt>h</tt>.
+ *  - <tt>compact!</tt> removes <tt>nil</tt>valued entries from that copy.
```

Comment:
```
irb(main):001:0> h = Hash.new
irb(main):002:0> h.compact!
=> nil
```

Reply:
Fixed.  I had originally used bang methods to avoid having to deal with intermediate values -- there would just be self.
The need now is to deal with intermediate values without referring to them by the variable names they don't have.
Also documented that the first method in the chain need not be in Hash.